onnxruntime/python/tools/transformers/benchmark_gpt2_LM/README.md

```diff
@@ -0,0 +1,100 @@
+**Benchmark tool**
+
+This can be used to test beam search on any gpt2 model. Both pytorch and onnx models can be tested.
+
+Other kind of models are not supported yet.
+
+Also, this script defaults to GPU. To test a model on CPU pass --device 'cpu' when running the model.
+
+
+
+**Requirements:**
+
+1. GPU machine. V100 was used to test this. 
+
+2. GPT2 ONNX Model that follows the regular GPT2 style of inputs and outputs.
+
+   Model should have 4 inputs, with the following dimensions:
+
+   **input_ids** `[batch_size,seq_len]`
+
+   **position_ids** `[batch_size,seq_len]`
+
+   **attention_mask** `[batch_size,total_seq_len]`
+
+   **past_0**`[2,batch_size,16,past_seq_len,64]`
+
+   **past_1** and so on with the same structure
+
+   And 2 outputs:
+
+   **logits**	type: `[batch_size,seq_len,50297]`
+
+   **present_0** type: `[2,batch_size,16,total_seq_len,64]`
+
+   **present_1** and so on with the same structure
+
+   past_0, attention_mask, logits, present_0 can be float16 or float32. inputs_ids and position_ids are int64.
+
+   
+
+3. Create python venv with all the packages in required_packages.txt
+
+   >  pip install -r requirements.txt
+
+   
+
+**Run the script:**
+
+```
+Test onnx:
+
+the model onnx_gpt2.onnx has to be converted using onnxruntime.transformers.convert_to_onnx if not already done.
+
+python .\model\main.py -t "onnx" -m "onnx_gpt2.onnx" --num_beams 2 --run_beam_search --length_penalty 1.6 --tokenizer gpt2  --input_file .\1K_queries.tsv --output_file .\1K_queries_onnx_output.tsv
```

Comment:
Could we organize the directory so that the command line could be like "python main.py ..."